### PR TITLE
Switch to tar.gz for Linux / MacOS

### PR DESCRIPTION
**Pull request by iamahern**
For Mac/Linux tar.gz is the more common archive format. Fixed gradlew executable settings and converted mac / linux defaults to tar.gz.

### CODE REVIEW

**Comment by iamahern** [comment]
According to the logs:
`Encrypted environment variables have been removed for security reasons.
See https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions`
That seems to have broken the integration tests.

**Comment by remmeier** [comment]
thx,

maybe it would also make sense to migrate or at least support https://hub.docker.com/r/lachlanevenson/k8s-helm/ to avoid the download of platform specific binaries.

I think the build failing is a security issue. Master just worked fine. Travis does not give away secure env variables for the pull request that come from outside the own repository. So I guess that tests should be disabled. Or rather swich to a local setup (most recent docker version have/soon will have kubernetes support).

**Comment by iamahern** [comment]
I think its a good test still of the end-to-end plugin. On travis you can have conditional steps that only run on master:
https://docs.travis-ci.com/user/conditional-builds-stages-jobs/

I'll add the conditional-master change to my pull requests.

On, the docker based helm client still - that is probably a good change, but it would require a more significant pull request. I would be minimally concerned that it would break some of your users.

-----

In a previous. pre-container, automation environment we had a restrictive firewall setup around the VM executing the gradle based installer that blocked all externally bound traffic. To solve this, we pre-bootstrapped the gradle dependencies offline and zipped them up to overcome these restrictions. Your current plugin could (theoretically) operate in this environment as it does not require external access to download the Helm client. The Kubernetes slave nodes would require external access to bootstrap Helm, but those may have a different firewall setup than the orchestration VM and container isolation is handled then by Kubernetes network policies.

Again, this was all pre-containers and I have trouble imagining this scenario for a Kubernetes orchestration system. As you master containers, the desire is to use containers to wrap your tool dependencies - but you never know if the change you are proposing would break someone.

**Comment by iamahern** [comment]
@remmeier - Do you mind completing the merge? We have a production system stuck on helm 2.7.x that requires the .tar.gz support to let users use. I would like to point users at the official plugin release. Working like a charm for our helm needs now.

I did some digging into supporting:
>   https://hub.docker.com/r/lachlanevenson/k8s-helm/

And it will not be straightforward. 

When you run the helm tool locally it automatically picks up environment variables / kubectl configuration from users system. It is not well documented what environmental configuration and user host system directories would need to be mounted into the container for this to run correctly.

Using various cloud providers tools for example, they may override the kubeconfig via KUBECONF environment variable, so the config file will be located on an arbitrary point in the host filesystem (EG: `KUBECONFIG=/Users/iamahern/.bluemix/plugins/container-service/clusters/my-cluster/kube-config-my-namespace.yml`

If you stick to the downloaded EXE your helm command inherit the correct kubernetes and the kubectl. 

Long story short, the cost of supporting multi-OS downloads is significantly lower than the headache of getting this to work correctly in a container would be.

**Comment by remmeier** [comment]
I merged and pushed to stable-kuberneates to make that release (0.2.20180621194407)

Currently I work on a terraform and azure plugin. There I'm started going the docker route to get the CLI tooling. It is a bit more effort since it needs the proper volume mounts as you also stated. But at the end so far it looks quite great because it no longer pollutes the system and the project is isolated from all external influences and brings along everything it needs. And I don't have to to build up all those platform specific download urls either. The created configurations get located somewhere in the build directory through volume mounts into the docker container. And a small shell script in the project root (helm.sh) could hide all this magic to have command line access just like with the native binaries.

**Comment by iamahern** [comment]
Cheers. Write me when you get something; I would be happy to help test in the context of my cloud environment.

**Comment by remmeier** [comment]
there is a first version now available on the master branch. It likely be refined a bit over the coming weeks.
